File: mpi-send-rcv.jl
using MPI
MPI.Init()
comm = MPI.COMM_WORLD
const rank = MPI.Comm_rank(comm)
const size = MPI.Comm_size(comm)
if rank == 0
 data11 = ones(3)*1
    data12 = ones(3)*2
    MPI.Send(data11,  1, 11, comm)
    MPI.Send(data12,  1, 12, comm)
elseif rank == 1
    data_recibida11 =zeros(3)
    data_recibida12 =zeros(3)
    MPI.Irecv!(data_recibida11, 0, 11, comm)
    MPI.Irecv!(data_recibida12, 0, 12, comm)  
end
MPI.Barrier(comm)
if rank == 1
println("En proceso ",rank,"  data_recibida11 =",data_recibida11)
println("En proceso ",rank,"  data_recibida12 =",data_recibida12)
end
MPI.Finalize() 

File: mpi-send-rcv.py
from mpi4py import MPI; import numpy as np ; import random
comm = MPI.COMM_WORLD
rank = comm.Get_rank()
if rank == 0:
   data11 = np.ones(3)*11
   data12 = np.ones(3)*12
   comm.send(data11, dest=1, tag=11)
   comm.send(data12, dest=1, tag=12)
elif rank == 1:
   data11_r = np.zeros(3)
   data12_r = np.zeros(3)
   data11_r = comm.recv(source=0, tag=11)
   data12_r = comm.recv(source=0, tag=12)
   print("En proceso %d, data11_recibido =%s"%(rank, data11_r))
   print("En proceso %d, data12_recibido =%s"%(rank, data12_r))
MPI.Finalize()
